Cirrious.FluentLayout/AdvancedFluentLayoutExtensions.cs

```diff
@@ -15,21 +15,41 @@ namespace Cirrious.FluentLayouts.Touch
 {
     public static class AdvancedFluentLayoutExtensions
     {
-		const float DefaultMargin = 0;
-		const float DefaultScale = 1;
+        const float DefaultMargin = 0;
+        const float DefaultScale = 1;
 
         public static FluentLayout AtTopOf(this UIView view, UIView parentView, nfloat? margin = null) =>
-			view.Top().EqualTo().TopOf(parentView).Plus(margin.GetValueOrDefault(DefaultMargin));
+            view.Top().EqualTo().TopOf(parentView).Plus(margin.GetValueOrDefault(DefaultMargin));
+
+        public static FluentLayout AtTopOfWithSafeArea(this UIView view, UIView parentView, nfloat? margin = null) =>
+            UIDevice.CurrentDevice.CheckSystemVersion(11, 0)
+                    ? view.Top().EqualTo().TopOf(parentView.SafeAreaLayoutGuide).Plus(margin.GetValueOrDefault(DefaultMargin))
+                    : view.AtTopOf(parentView, margin);        
 
         public static FluentLayout AtLeftOf(this UIView view, UIView parentView, nfloat? margin = null) =>
 			view.Left().EqualTo().LeftOf(parentView).Plus(margin.GetValueOrDefault(DefaultMargin));
 
+        public static FluentLayout AtLeftOfWithSafeArea(this UIView view, UIView parentView, nfloat? margin = null) =>
+            UIDevice.CurrentDevice.CheckSystemVersion(11, 0)
+                    ? view.Left().EqualTo().LeftOf(parentView.SafeAreaLayoutGuide).Plus(margin.GetValueOrDefault(DefaultMargin))
+                    : view.AtLeftOf(parentView, margin);
+
         public static FluentLayout AtRightOf(this UIView view, UIView parentView, nfloat? margin = null) =>
 			view.Right().EqualTo().RightOf(parentView).Minus(margin.GetValueOrDefault(DefaultMargin));
 
+        public static FluentLayout AtRightOfWithSafeArea(this UIView view, UIView parentView, nfloat? margin = null) =>
+            UIDevice.CurrentDevice.CheckSystemVersion(11, 0)
+                    ? view.Right().EqualTo().RightOf(parentView.SafeAreaLayoutGuide).Plus(margin.GetValueOrDefault(DefaultMargin))
```

Comment:
In order to be used as `AtRightOf` this should use `Minus` for the margin:

```suggestion
                    ? view.Right().EqualTo().RightOf(parentView.SafeAreaLayoutGuide).Minus(margin.GetValueOrDefault(DefaultMargin))
```